kong/conf_loader/init.lua

```diff
@@ -1861,6 +1861,13 @@ local function load(path, custom_conf, opts)
 
   log.verbose("prefix in use: %s", conf.prefix)
 
+  -- set unix domain socket file absolute path
+  local unix_socket_path = getenv("KONG_UNIX_SOCKET_PATH")
```

Comment:
Why is this needed?

Reply:
See the discussion in PR #9337, we can customize the unix socket path by env.

Reply:
I am not convinced that this is a good idea, and I will block this PR until the concern is clearly addressed (see: https://github.com/Kong/kong/pull/9337#pullrequestreview-1143550684).

Why does Kong needs a user configurable socket directory in order to function correctly?

Reply:
It has a default value (`ngx.config.prefix()`), so the user does not need to know it.

I think this env is only for internal usage, and it is a POC for plan B in the ticket.

Reply:
@dndx We need it for testing, see https://github.com/Kong/kong/pull/9337#issuecomment-1247893735